wordpress-mobile/WordPress-iOS
# Feature/9687 formattable content as protocol

Fixes #9687 

This PR implements the [changed made on WordPressKit](https://github.com/wordpress-mobile/WordPressKit-iOS/pull/20), where `FormattableContent` (aka `NotificationBlock` was converted into a protocol.

This allows us to encapsulate code specific for a kind of content (i.e. user or comment) in their own classes.
This also allow us to add new behavior (or content kinds) in the future by creating a new class, without modifying existing ones.

What's next:
- We would like to do a similar change to Ranges.
- We would like to simplify the controllers (`Notifications` and `NotificationDetail`) by encapsulating some of the blocks related code into their specialized `ContentGroups`.

To test:
- Build and run
- Check that all notifications are displayed as they should.
- Check that Activity log displays with formatting (just some italics for now, it will be improved in the future)
